ts/components/LabelledItem/Icon.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from "react";
+import { ImageSourcePropType, ImageStyle, Image } from "react-native";
+import { isString } from "lodash";
+import { IconProps } from "react-native-vector-icons/Icon";
+import IconFont from "../ui/IconFont";
+import variables from "../../theme/variables";
+
+// Style type must be generalized so as to support both text icons and image icons
+type StyleType = IconProps["style"] & ImageStyle;
+
+type Props = {
+  icon?: string | ImageSourcePropType;
+  iconColor?: string;
+  iconStyle?: StyleType;
+  accessibilityLabelIcon?: string;
+  onPress?: () => void;
+};
+
+export const Icon: React.FC<Props> = ({
+  icon,
+  iconColor,
+  iconStyle,
+  onPress,
+  accessibilityLabelIcon
+}): React.ReactElement => (
+  <>
+    {icon &&
```

Comment:
This check should be left to the consumer: if there is no `icon` there should be no way to instantiate an `Icon` component, and being the characterizing property of an Icon it should not be optional.